### ISSUE TITLE
redraw doesn't work if separator changed

### ISSUE THREAD

**Issue by MichaelLonin** [glennreyes/react-countup#151]
If new separator provided redraw={true} doesn't trigger the animation. start() do but separator still shows old value.

**Comment by glennreyes** [comment]
@MichaelLonin Thanks for filing this issue. Could you please provide a CodeSandbox example of this bug?

**Comment by mmarkelov** [comment]
@glennreyes I suppose issue about that counter doesn't change state with changing separator:
```javascript
import React, { useState } from "react";
import ReactDOM from "react-dom";
import CountUp from 'react-countup';


import "./styles.css";

function App() {
  const [separator, setSeparator] = useState('')

  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <button onClick={() => setSeparator(' ')}>Change separator</button>  
      <CountUp end={10000} duration={10} separator={separator} redraw />
    </div>
  );
}

const rootElement = document.getElementById("root");
ReactDOM.render(<App />, rootElement);

```

I don't see any particular cases for this.

@MichaelLonin can you explain your purpose?

**Comment by Kenny407** [comment]
Hello. I have a use-case for this. 

I'm using this component to count time, specifically with a user friendly display in seconds. e.g.:
If I display a number in time (from 1 to 59), the user will read "1 second" or "1s". If the count goes above 60 to 3599 then the user will view "1 min", "2 min" etc up to "59 min", and consequently for hours, days, etc.

I found the problem and it's because in the [componentDidUpdate](https://github.com/glennreyes/react-countup/blob/master/src/CountUp.js#L68) it's the instance is only recreated if the duration or the start change.

**Comment by mmarkelov** [comment]
@Kenny407 that's make sense

**Comment by mmarkelov** [comment]
should be fixed in **v4.2.3**
